examples/custom-metrics/custommetrics_example/server_app.py

```diff
@@ -0,0 +1,74 @@
+"""custommetrics_example: A Flower / TensorFlow app for custom metrics."""
+
+import numpy as np
+
+from flwr.common import Context
+from flwr.server import ServerApp, ServerAppComponents, ServerConfig
+from flwr.server.strategy import FedAvg
+
+
+# Define metrics aggregation function
+def average_metrics(metrics):
+    # pylint: disable=C0301
+    """Aggregate metrics from multiple clients by calculating mean averages.
+
+    Parameters
+    ----------
+    metrics : list
+        A list containing tuples, where each tuple represents metrics for a client.
+        Each tuple is structured as (num_examples, metric), where:
+        - num_examples (int) : The number of examples used to compute the metrics.
+        - metric (dict) : A dictionary containing custom metrics provided as
+                          `output_dict` in the `evaluate` method from `client.py`.
+
+    Returns
+    -------
+    dict
+        A dictionary with the aggregated metrics, calculating mean averages.
+        The keys of the dictionary represent different metrics, including:
+        - 'accuracy': Mean accuracy calculated by TensorFlow.
+        - 'acc': Mean accuracy from scikit-learn.
+        - 'rec': Mean recall from scikit-learn.
+        - 'prec': Mean precision from scikit-learn.
+        - 'f1': Mean F1 score from scikit-learn.
+
+        Note: If a weighted average is required, the `num_examples` parameter can be
+        leveraged.
+
+        Example:
+            Example `metrics` list for two clients after the last round:
+            [(10000, {'prec': 0.108, 'acc': 0.108, 'f1': 0.108, 'accuracy': 0.1080000028014183, 'rec': 0.108}),
+            (10000, {'f1': 0.108, 'rec': 0.108, 'accuracy': 0.1080000028014183, 'prec': 0.108, 'acc': 0.108})]
```

Comment:
I wonder if we should keep all this now that we are updating the example. the returns description is subject to what the `evaluate()` method at the client is.

Reply:
I suppose this is useful to emphasize that we're returning different metrics, one of which is calculated using different libraries. It's useful information to users.